### ISSUE TITLE
The foundation variant puts folders in app/frontend instead of app/frontend/stylesheets

### ISSUE THREAD

**Issue by eoinkelly** [ackama/rails-template#123]
For example, this is what the layout of `app/frontend` looks like currently if you choose to install foundation + foundation layout variants:

```
app/frontend
├── application.scss
├── base
├── channels
│   ├── consumer.js
│   └── index.js
├── components
├── foundation
│   ├── base
│   │   ├── base.scss
│   │   ├── buttons.scss
│   │   ├── containers.scss
│   │   ├── forms.scss
│   │   └── typography.scss
│   ├── components
│   │   ├── _footer.scss
│   │   ├── _header.scss
│   │   ├── _navigation.scss
│   │   ├── _search-bar.scss
│   │   ├── _social-media.scss
│   │   └── header.js
│   ├── foundation_and_overrides.scss
│   ├── index.js
│   └── settings.scss
├── images
│   └── logo.svg
├── layouts
├── packs
│   └── application.js
└── stylesheets
    ├── _elements.scss
    └── application.scss

10 directories, 21 files
```

All styles should instead be under `app/frontend/stylesheets`

**Comment by joshmcarthur** [comment]
I personally prefer the layout as it is. If you take a look at that directory tree, you'll see that it includes more than stylesheets. It also includes an `index.js` entrypoint, and could include other files, like foundation-specific iconography or images. I think that moving this directory to stylesheets wouldn't necessarily be correct either, since those other files would have to be placed elsewhere.

I think generally, we're running up against a Rails asset pipeline vs webpack ethos issue here - Rails asset pipeline wants things in discrete directories so it can easily point different tools at each type of file to preprocess - webpack's method is to namespace as much as you want, but within those namespaces to dump all types of files into those directories, and let webpack's loaders figure out how to get them into the compiled assets.

I suspect the latter approach is a bit more compatible with the dev experience others will have from things like create-react-app (which also dumps all file types in `src`), and that it's not necessarily bad to have different types of files in a folder at the top-level if it keeps all those files that are intended for a single purpose/library in a single place.

**Comment by joshmcarthur** [comment]
One thing I forgot to mention -I believe the intention of the folder structure as it stands was to make the 'foundation' directory resemble 'app/frontend' so that there was a clear structure to those files - you'll notice how the 'foundation' directory has a JS entrypoint, `base`, `components` etc., just like `app/frontend` has.

**Comment by Rabid-Dan** [comment]
I've used this structure a handful of times recently and didn't have a ton of problems with it personally. 
I _did_ end up doing a few double takes on though on NZX which is bootstrap rather than foundation based, and where bootstrap had been added after generation rather than as a variant. The bootstrap overrides were placed in the stylesheets directory (which feels like a reasonable place to put them without the context of how and why the foundation variant generates directories tbh) rather than a dedicated `bootstrap` directory like they are for the foundation variant which kept tripping me up when I was trying to remember whether custom styles lived in `frontend/base, frontend/components, frontend/layouts` or `frontend/stylesheets/base, frontend/stylesheets/components, frontend/stylesheets/layouts`.

I think that case alone isn't necessarily a 100% argument for shifting base/components/layouts into the stylesheets directory but it definitely highlighted to me how there could be a bit of cognitive dissonance, particularly for new developers, when trying to decide where styles should go.
It also reminded me that during the times I've used the foundation variant the `stylesheets` directory has always felt kind of redundant given that the only thing the ends up living there is the application.scss file.
